kernel_gateway/jupyter_websocket/__init__.py

```diff
@@ -0,0 +1,65 @@
+from ..base.handlers import default_handlers as default_base_handlers
+from ..services.activity.handlers import ActivityHandler
+from ..services.kernels.pool import KernelPool
+from ..services.kernels.handlers import default_handlers as default_kernel_handlers
+from ..services.kernelspecs.handlers import default_handlers as default_kernelspec_handlers
+from ..services.sessions.handlers import default_handlers as default_session_handlers
+from .handlers import default_handlers as default_api_handlers
+from notebook.utils import url_path_join
+
+class JupyterWebsocketPersonality(object):
+    '''Personality for standard websocket functionality, registering
+    endpoints that are part of the Jupyter Kernel Gateway API
+    '''
+    def __init__(self, kernelspec):
```

Comment:
It seems odd to have `kernelspec` be the one and only constructor arg. I think I know why you're passing it for notebook http server, but I only see `None` ever passed when creating a personality, so maybe I'm missing how it's intended used. 

Still, it seems like there should be a more elegant way. Would passing the gatewayapp instance to the constructor and accessing its attributes help?